scheduler/src/cook/scheduler/scheduler.clj

```diff
@@ -1554,6 +1555,7 @@
   (persist-mea-culpa-failure-limit! conn mea-culpa-failure-limit)
 
   (let [{:keys [match-trigger-chan rank-trigger-chan]} trigger-chans
+        match-trigger-chan-mult (async/mult match-trigger-chan)
```

Comment:
Looks good. One question. Did you do a before-after comparison? 

Before, this log line (in match-jobs-event): should print roughly one second apart across each pool:
        (log/info "In" pool-name "pool, starting offer matching")

With the change, you should see them all start at the same time, within tens of milliseconds. This should be true even if some of them are really slow; they'll just skip the chime and get the next one. (E.g., during testing, make this pause a variable time. e.g., (+ 2 (% (hash pool) 10)) seconds and make sure that nothing seems to pause an excessive amount.

Reply:
the logs looked as expected in terms of chiming at the same time

not sure how to test the very long scheduling time. open to suggestions

Reply:
Within the body of the function, put in a Thread/sleep with a delay calculated based on the pool name.